test/implementations/lazyMintErc20Pay.ts

```diff
@@ -151,6 +151,21 @@ async function shouldControlValidMintingErc20Pay(): Promise<void> {
     expect(await this.token.balanceOf(addrs[0].address)).to.equal(1);
   });
 
+  it('reduces total supply on burn', async function () {
+    const tokenId = await mintFromImplErc20Pay(this.token, addrs[0].address);
+    await this.token.connect(addrs[0])['burn(uint256)'](tokenId);
+    expect(await this.token.totalSupply()).to.equal(0);
+  });
+
+  it('reduces total supply on burn and does not reuse Id', async function () {
+    const tokenId = await mintFromImplErc20Pay(this.token, addrs[0].address);
+    await this.token.connect(addrs[0])['burn(uint256)'](tokenId);
+
+    const newTokenId = await mintFromImplErc20Pay(this.token, addrs[0].address);
+    expect(newTokenId).to.equal(tokenId.add(1));
+    expect(await this.token.totalSupply()).to.equal(1);
```

Comment:
Can we add a validation that `tokenId` is not equal to `newTokenId` as well?

Reply:
It's already doing it on line 165. New token Id must be equal to `tokenId + 1`.